components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/services/ConditionalAuthenticationMgtService.java

```diff
@@ -44,7 +44,7 @@ public class ConditionalAuthenticationMgtService extends AbstractAdmin {
      */
     public String[] getAllAvailableFunctions() {
 
-        List<String> jsFunctions = Stream.of("executeStep", "selectAcrFrom", "sendError", "Log.info", "require").collect
+        List<String> jsFunctions = Stream.of("executeStep", "selectAcrFrom", "sendError", "Log.info", "require", "secrets.getSecret").collect
```

Comment:
so you seems to be defining the secret function for graal and nashron.
This is added in the context of nashron only which is quite misleading as this is a separate class that doesn't depends on that JS engine implementation

Reply:
resolved with https://github.com/wso2/carbon-identity-framework/pull/5705/commits/900b6de1b94d83e89fd7f62562b6d421f27e94d6